aten/src/ATen/native/cuda/BatchLinearAlgebra.cpp

```diff
@@ -2883,6 +2891,8 @@ static void apply_lu_solve_batched_magma(const Tensor& b, const Tensor& lu, cons
       "Calling torch.lu_solve on a CUDA tensor requires compiling ",
       "PyTorch with MAGMA. Please rebuild with MAGMA.");
 #else
+  TORCH_INTERNAL_ASSERT(batchCount(b) == batchCount(lu), "batch_size of b and lu must be the same");
+  TORCH_INTERNAL_ASSERT(batchCount(lu) == batchCount(pivots.unsqueeze(-1)), "batch_size of lu and pivots must be the same");
```

Comment:
Just another note. I just realised that we can implement a faster `batchCount` doing `t.dim() > 2 ? t.numel() / t.stride(-3) : 1` when we have a batch of column or row-major matrices.